analyzers/src/SonarAnalyzer.CSharp/Rules/UnnecessaryUsings.cs

```diff
@@ -74,8 +81,12 @@ private static void CheckUnnecessaryUsings(SonarSyntaxNodeReportingContext conte
             foreach (var usingDirective in usingDirectives)
             {
                 // This will create some FNs but will kill noise from FPs.
-                // For more info see issue: https://github.com/SonarSource/sonar-dotnet/issues/5946
-                if (usingDirective.GetFirstToken().IsKind(SyntaxKind.GlobalKeyword))
+                // For more info see issues:
+                //  - https://github.com/SonarSource/sonar-dotnet/issues/5946
+                //  - https://github.com/SonarSource/sonar-dotnet/issues/7959
+                if (usingDirective.GetFirstToken().IsKind(SyntaxKind.GlobalKeyword)
```

Comment:
The logic for this is OK. And If I am not mistaken the logic for global using is not correct. That one should do a `continue` instead of a `return`. Since it can happen that only some of the using collected are global and not all. Since we are literally changing the line, can you add UTs for it and fix it?

Reply:
Nice catch!

Reply:
For both, `global using` and `_Imports.razor`/`_ViewImports.cshtml`, it needs to be `continue`.

Reply:
Why?

Reply:
`_Imports.razor` and `_ViewImports.cshtml` will be included in the generated content before any `using` of the current razor file. With a `return` we won't reach the `using`s from the razor file.

Reply:
Of course. 🤦 I got misled by the UTs. Can you add a UT that actually showcases this? Because if somebody just looks at them like I did and forgets the fact that the content is merged with other razor files, it might think we only ignore the usings in that 1 file only. But we actually want to ignore them from wherever we map back there.

Reply:
The UTs should all there 😉

Reply:
Now I see it, but since you do git magic for every push and everything is in 1 commit, it is unbelievably difficult to see what were the changes between my reviews.